Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -393,4 +391,30 @@ namespace AzToolsFramework::ComponentModeFramework
         // Send a list of selected and deselected entities to the switcher to deal with updating the switcher view
         UpdateSwitcherOnEntitySelectionChange(newlySelectedEntities, newlyDeselectedEntities);
     }
+
+    void ComponentModeSwitcher::AfterUndoRedo()
+    {
+        // Wait one frame for the undo stack to actually be updated.
+        QTimer::singleShot(
+            0,
+            nullptr,
+            [this]()
+            {
+                auto* toolsApplicationRequests = AzToolsFramework::ToolsApplicationRequestBus::FindFirstHandler();
+                const auto& selectedEntityIds = toolsApplicationRequests->GetSelectedEntities();
+
+                if (selectedEntityIds.size() == 1)
```

Comment:
question: What happens if there are multiple components selected? Will this fail in that case? Do we need to add support for that in a follow-up PR?

Reply:
I did this because firstly we only care about the Undo action here if an entity is actually selected. The switcher doesn't need to check anything if no components are selected because AfterEntitySelectionChange should handle those scenarios. If an entity is selected, it seems like you can only disable/remove components when only one is selected; I guess the check isn't needed but I didn't want to risk any unhandled behaviour

Reply:
Hmm I see 🤔 So if I say have three entities selected (all with the same components) and I deselect (click off) and then press undo, `AfterEntitySelectionChange` is still called? And is it also called with redo as well?

Reply:
Yes AfterEntitySelectionChange is still called when the entity selection changes due to undo/redo